packages/rich-text-types/src/validator/index.ts

```diff
@@ -0,0 +1,114 @@
+import { BLOCKS } from '../blocks';
+import { INLINES } from '../inlines';
+import { CONTAINERS, LIST_ITEM_BLOCKS, TOP_LEVEL_BLOCKS } from '../schemaConstraints';
+import { Document, Text } from '../types';
+import { ObjectAssertion } from './assert';
+import { NodeAssertion, Node, HyperLinkAssertion, assert, assertLink, VOID_CONTENT } from './node';
+import { Path } from './path';
+import { assertText } from './text';
+
+export type ValidationError = {
+  name: string;
+  type?: string;
+  value?: Record<string, any> | string | number | boolean | null;
+  min?: number | string;
+  max?: number | string;
+  details?: string | null;
+  path?: (string | number)[];
+  contentTypeId?: string | string[];
+  nodeType?: string;
+  customMessage?: string;
+  expected?: string[];
+};
+
+const assertInlineOrText = assert([...Object.values(INLINES), 'text'].sort());
+
+const assertList = assert([BLOCKS.LIST_ITEM]);
+const assertVoidEntryLink = assertLink('Entry', VOID_CONTENT);
+const assertTableCell = assert(
+  () => ({
+    nodeTypes: [BLOCKS.PARAGRAPH],
+    min: 1,
+  }),
+  (data, path) => {
+    const $ = new ObjectAssertion(data, path);
+
+    $.noAdditionalProperties(['colspan', 'rowspan']);
+    $.number('colspan', true);
+    $.number('rowspan', true);
+
+    return $.errors;
+  },
+);
+
+const nodeValidator: Record<Node['nodeType'], NodeAssertion<any>> = {
+  [BLOCKS.DOCUMENT]: assert(TOP_LEVEL_BLOCKS),
+  [BLOCKS.PARAGRAPH]: assertInlineOrText,
+  [BLOCKS.HEADING_1]: assertInlineOrText,
+  [BLOCKS.HEADING_2]: assertInlineOrText,
+  [BLOCKS.HEADING_3]: assertInlineOrText,
+  [BLOCKS.HEADING_4]: assertInlineOrText,
+  [BLOCKS.HEADING_5]: assertInlineOrText,
+  [BLOCKS.HEADING_6]: assertInlineOrText,
+  [BLOCKS.QUOTE]: assert(CONTAINERS[BLOCKS.QUOTE]),
+  [BLOCKS.EMBEDDED_ENTRY]: assertVoidEntryLink,
+  [BLOCKS.EMBEDDED_ASSET]: assertLink('Asset', VOID_CONTENT),
+  [BLOCKS.EMBEDDED_RESOURCE]: assertLink('Contentful:Entry', VOID_CONTENT),
+  [BLOCKS.HR]: assert(VOID_CONTENT),
+  [BLOCKS.OL_LIST]: assertList,
+  [BLOCKS.UL_LIST]: assertList,
+  [BLOCKS.LIST_ITEM]: assert([...LIST_ITEM_BLOCKS].sort()),
+  [BLOCKS.TABLE]: assert(() => ({
+    nodeTypes: [BLOCKS.TABLE_ROW],
+    min: 1,
+  })),
+  [BLOCKS.TABLE_ROW]: assert(() => ({
+    nodeTypes: [BLOCKS.TABLE_CELL, BLOCKS.TABLE_HEADER_CELL],
+    min: 1,
+  })),
+  [BLOCKS.TABLE_CELL]: assertTableCell,
+  [BLOCKS.TABLE_HEADER_CELL]: assertTableCell,
+  [INLINES.HYPERLINK]: new HyperLinkAssertion(),
+  [INLINES.EMBEDDED_ENTRY]: assertVoidEntryLink,
+  [INLINES.EMBEDDED_RESOURCE]: assertLink('Contentful:Entry', VOID_CONTENT),
+  [INLINES.ENTRY_HYPERLINK]: assertLink('Entry', ['§']),
```

Comment:
what is this `§`??

Reply:
@YvesRijckaert / @chrishelgert ☝🏻

Reply:
let me check 👁️

Reply:
It's supposed to be `text` just like the other links I think. 

But the fact this didn't fail any test is a bit alarming 🙈 . Could you add some bitte 🙇🏻 ?

Reply:
typo from copying over, will extend the test to cover this

Reply:
test extended 👍